bin/libs/string.ts

```diff
@@ -344,3 +344,16 @@ export function itemInList(stringList: string, stringToFind?: string, separator:
   }
   return stringList.split(separator).includes(stringToFind);
 }
+
+export function escapeDollar(str: string): string | undefined {
+  if (str === null || str === undefined)
+    return undefined;
+  return str.replace(/[$]/g, '\\$&');
```

Comment:
can you explain what the ampersand '&' does here?
I'm not familiar with it and a quick search didnt tell me why it is here.

Reply:
Explanation [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_string_as_the_replacement).

`$&` = inserts the matched substring.

It is better to show on the second replacement:
`return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');`: any character from `[]` found, replace by backslash and that character.

The first replace is just for one char, but the replacement is in general form. It can be changed to `str.replace(/[$]/g, '\\$');` if it makes the code better to read. This or that, it will work the same.

Reply:
okay now i understand. i thought \$& was a shell thing, i didn't realize it was a javascript thing.